app/models/foreman_azure_rm/azure_rm_compute.rb

```diff
@@ -0,0 +1,108 @@
+module ForemanAzureRM
+  class AzureRMCompute
+    attr_accessor :sdk
+    attr_accessor :azure_vm
+
+    delegate :name, to: :azure_vm, allow_nil: true
+
+    def initialize(azure_vm: ComputeModels::VirtualMachine.new, sdk: sdk)
+      require 'pry';binding.pry unless azure_vm.is_a? ComputeModels::VirtualMachine
+      @azure_vm = azure_vm
+      @sdk = sdk
+    end
+
+    # All the earlier properties are no longer present here, they've
+    # been wrapped directly inside view.
+    # Only foreman related properties would be taken care of here.
+
+    def id
+      @azure_vm.id
+    end
+
+    def persisted?
+      !!identity && !!id
+    end
+
+    def vm_size
+      @azure_vm.hardware_profile.vm_size
+    end
+
+    def wait_for(_timeout = 0, _interval = 0, &block)
+      instance_eval(&block)
+      return true
+    end
+
+    def ready?
+      vm_status == 'running'
+    end
+
+    def state
+      vm_status
+    end
+
+    def vm_status
+      sdk.check_vm_status(@azure_vm.resource_group, name)
+    end
+
+    def interfaces_attributes=(attrs)
```

Comment:
Why do we ignore changes to interfaces?

Reply:
I had kept the implementation similar to the [earlier code](https://github.com/theforeman/foreman_azure_rm/blob/master/app/models/concerns/fog_extensions/azurerm/server.rb#L77). How should we set value to interface attributes then?

Reply:
We already do it when we call for a new vm. I wonder if we get here at all. Please check it out.